### DISCUSSION TITLE
How to tell if a geometry is disjoint?

### DISCUSSION THREAD

**Discussion by BarbourSmith** [sgenoud/replicad#166] [Modelling help] [answered]
I have an interesting issue.

I'm working on automatically nesting parts to fit on a sheet to be cut out, and by and large it's working great. 

<img width="546" alt="image" src="https://github.com/user-attachments/assets/d0959736-8da3-4421-9322-368652ba3974">

The issue is that when I have a single part (in this case created a rectangle which was extruded) which is then cut into three pieces by another part using .cut 

<img width="262" alt="image" src="https://github.com/user-attachments/assets/938174e5-bc4a-4ca4-89a7-6d7d74560e71">

The issue is that when we get to the final nesting step we want to treat each of these three separate disjoint parts as unique and lay them out separately. Where I'm stuck is that I'm not sure how to tell that these parts are no-longer connected together.

Is there a way that I can see that this single geometry is actually three disjoint geometries and treat them that way? IE in this case the three rectangles would be handled the same as if they had been created with three rectangles extruded individually instead of one rectangle split into three with .cut?

**Comment by sgenoud** [comment, marked as answer]
You can list the solids in the shape, with a function like that:

```js
function getSolids(compound) {
  return Array.from(iterTopo(compound.wrapped, "solid"), (s) => new Solid(s));
}
```

This is using the mechanism that I use to list all edges or faces of a shape exposed as a function. 

You can see it [used here](https://studio.replicad.xyz/workbench#code=UEsDBAoAAAAIAHE0O1mp3WJ%252BSwEAAI0CAAAHAAAAY29kZS5qc32Q0WvCMBDG3%252FtXHH1KpFTHmA%252BKY2Ovg4nbm%252FgQ06uGpUlIU1RK%252F%252FddW6vC5gqF5O6%252B3%252FflxqMRvISTwwxzqKE92RxU4awPLPbotJIiizk0MFze1RZG4ygaD1Kob1pN25PWlIFwKqD%252Fss4mkHlxeFNeakzg02qV9aWVrUyG2QplEGankWwWF6N5FOGxDQI9jyKKSoel8KIoaa5uaCKvjAzKGthh6MAlk5biE5dDHQHRQuUNvHovTmnubcGGVJfB9OCFoxUkEJctIuYJsJLD4hkMHvq8dOfzqLlEOoeBi38hlGG9ZR9XVoFC%252FvVKNk1gytPyG4Pcf5ilFgYZT%252FEYfJVRl4wGiLuHeCLE5D7jkafkz%252Bi%252FgZV74bDd3HVXjtrXJa3pCN0T2q8bn51V68kmOdel1dbPIN7qCuO%252B2CT%252FKB9%252BKXce0dxKN91qfwBQSwECFAAKAAAACABxNDtZqd1ifksBAACNAgAABwAAAAAAAAAAAAAAAAAAAAAAY29kZS5qc1BLBQYAAAAAAQABADUAAABwAQAAAAA%253D).


Note I am not 100% sure of my answer (there might be edge cases about it in the underlying library that I am not aware of). Tell me if you find them.
